apps/mobile-wallet/src/store/addresses/addressesStorageUtils.ts

```diff
@@ -27,15 +28,23 @@ import { persistAddressesSettings } from '~/utils/addresses'
 export const importAddresses = async (walletId: WalletMetadata['id'], addressesMetadata: AddressMetadata[]) => {
   const addressHashes = []
 
-  for (const { index, label, color, isDefault } of addressesMetadata) {
-    const newAddressData = keyring.generateAndCacheAddress({ addressIndex: index })
-    const newAddress = { ...newAddressData, settings: { label, color, isDefault } }
+  await initializeKeyringWithStoredWallet()
 
-    await persistAddressesSettings([newAddress], walletId)
-    store.dispatch(newAddressGenerated(newAddress))
+  try {
+    for (const { index, label, color, isDefault } of addressesMetadata) {
+      const newAddressData = keyring.generateAndCacheAddress({ addressIndex: index })
+      const newAddress = { ...newAddressData, settings: { label, color, isDefault } }
 
-    addressHashes.push(newAddress.hash)
-  }
+      await persistAddressesSettings([newAddress], walletId)
+      store.dispatch(newAddressGenerated(newAddress))
+
+      addressHashes.push(newAddress.hash)
+    }
 
-  store.dispatch(syncLatestTransactions(addressHashes))
+    store.dispatch(syncLatestTransactions(addressHashes))
+  } catch (e) {
+    throw new Error(getHumanReadableError(e, ''))
+  } finally {
+    keyring.clearAll()
```

Comment:
Just thinking about this now, what about renaming the `clearAll` method to simply `clear`, as we essentially clear the whole keyring and we don't have other more granular `clearSomething` methods?

Reply:
We have the `keyring.clearCachedSecrets`. Once the mobile wallet changes get merged, there won't be a need for this anymore and we can indeed remove it in favor of just `clear`. Since we are not caching secrets for mobile anymore, the code for the keyring can be reverted to the more simplified one. To keep this PR short, I did not go ahead with these changes.

Reply:
Sounds good!